nexus/src/app/sagas/instance_create.rs

```diff
@@ -281,34 +276,51 @@ impl NexusSaga for SagaInstanceCreate {
             ));
             subsaga_append(
                 "external_ip",
-                subsaga_builder,
+                subsaga_builder.build()?,
                 &mut builder,
                 repeat_params,
                 i,
             )?;
         }
 
+        for (i, disk) in params.create_params.disks.iter().enumerate() {
+            if let params::InstanceDiskAttachment::Create(create_disk) = disk {
+                let subsaga_name =
+                    SagaName::new(&format!("instance-create-disk-{i}"));
+                let subsaga_builder = DagBuilder::new(subsaga_name);
+                subsaga_append(
+                    "create-disk",
+                    SagaDiskCreate::make_saga_dag(
+                        &disk_create::Params {
+                            serialized_authn: params.serialized_authn.clone(),
+                            project_id: params.project_id.clone(),
+                            create_params: create_disk.clone(),
+                        },
+                        subsaga_builder,
+                    )?,
+                    &mut builder,
+                    create_disk,
+                    i,
+                )?;
+            }
+        }
+
         // See the comment above where we add nodes for creating NICs.  We use
         // the same pattern here.
         for i in 0..(MAX_DISKS_PER_INSTANCE as usize) {
             let disk_params =
                 DiskParams { saga_params: params.clone(), which: i };
             let subsaga_name =
-                SagaName::new(&format!("instance-create-disk{i}"));
+                SagaName::new(&format!("instance-attach-disk-{i}"));
             let mut subsaga_builder = DagBuilder::new(subsaga_name);
-            subsaga_builder.append(Node::action(
-                "create_disk_output",
-                format!("CreateDisksForInstance-{i}").as_str(),
-                CREATE_DISKS_FOR_INSTANCE.as_ref(),
-            ));
             subsaga_builder.append(Node::action(
                 "attach_disk_output",
                 format!("AttachDisksToInstance-{i}").as_str(),
                 ATTACH_DISKS_TO_INSTANCE.as_ref(),
             ));
```

Comment:
The attach behavior is still wrong. As per the comment below it only needs to operate on the datastore state (for now) which means it'll be a bit closer to the NIC behavior defined elsewhere in this file. I can go through and update that while I'm here. 

https://github.com/oxidecomputer/omicron/blob/09447c20716761f762d00f0cf71dab1d4486e3e7/nexus/src/app/instance.rs#L759-L761